Src/Support/Google.Apis.Auth.Tests/OAuth2/GoogleCredentialTests.cs

```diff
@@ -476,5 +476,40 @@ private void AssertContainsQuotaProject(IEnumerable<KeyValuePair<string, IEnumer
             string quotaProject = Assert.Single(headerValues.Value);
             Assert.Equal(expectedValue, quotaProject);
         }
+
+        [Fact]
+        public void TestImpersonatedCredential()
+        {
+            var sourceCredential = GoogleCredential.FromJson(DummyServiceAccountCredentialFileContents);
+            var delegates = new string[] {"delegate"};
+            var scopes = new string[] {"scope"};
+            var targetPrincipal = "foo";
+            var credential = sourceCredential.Impersonate(targetPrincipal, delegates, scopes, 4000);
+            var impersonatedCredential = credential.UnderlyingCredential as ImpersonatedCredential;
```

Comment:
I'd suggest a cast rather than `as` - it provoke a clearer failure if the underlying credential *isn't* an ImpersonatedCredential.

Reply:
done